rclcpp/include/rclcpp/node_interfaces/node_graph.hpp

```diff
@@ -61,6 +61,11 @@ class NodeGraph : public NodeGraphInterface
 
   RCLCPP_PUBLIC
   virtual
+  std::vector<std::string>
```

Comment:
I am not sure if the return type should be a vector. Since we don't need indexed access maybe a list would be more appropriate? A set would also be an option but since we are not clear if the names need to be unique that is probably too restrictive.